components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1482,6 +1485,72 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
         await this.guardAccess({ kind: "team", subject: team, members }, op);
     }
 
+    async getProviderRepositoriesForUser(params: { provider: string }): Promise<ProviderRepository[]> {
+        this.checkAndBlockUser("getProviderRepositoriesForUser");
+        // Note: this operation is per-user only, hence needs no resource guard
+        return [];
+    }
+
+    public async createProject(params: CreateProjectParams): Promise<Project> {
+        this.checkUser("createProject");
```

Comment:
Why not `checkAndBlockUser`? IMO we block all modifying/creating actions on block.